### PR TITLE
[Websearch] Implement running actual websearch action

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Fixes https://github.com/dust-tt/tasks/issues/780

Implement running the action for legacy agents & multi-actions, adding all relevant events and events generator.

Does **not** implement action rendering, planned for the following PR.
image

Risks
---
N/A (gated)

Deploy Plan
---
- Create prod secret for DUST_MANAGED_SERP_API_KEY
- deploy front

Note: there is *no* model change in this PR, the migration-ack is just a modified import path

### CODE REVIEW

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import type { WebsearchActionType } from "@dust-tt/types/dist/front/assistant/actions/websearch";
+import type { WebsearchActionType } from "@dust-tt/types/dist";
```
Comment (spolu) [inline front/components/assistant/conversation/WebsearchAction.tsx:1]: This solo import looks fishy?

File: front/lib/api/assistant/actions/websearch.ts [hunk 2]
```diff
@@ -1,10 +1,95 @@
...
+    this.functionCallName = blob.functionCallName;
+    this.step = blob.step;
+  }
+
+  renderForModel(): ModelMessageType {
+    let content = "";
+
+    // Note action.output can be any valid JSON including null.
+    content += `WEBSEARCH OUTPUT:\n`;
+    content += `${JSON.stringify(this.output, null, 2)}\n`;
```
Comment (spolu) [inline front/lib/api/assistant/actions/websearch.ts:62]: I think we can have a special case if this.output is null saying (web search failed) ?

File: front/lib/api/assistant/actions/websearch.ts [hunk 3]
```diff
@@ -1,10 +1,95 @@
...
+      arguments: JSON.stringify({ query: this.query }),
+    };
+  }
+
+  renderForMultiActionsModel(): FunctionMessageTypeModel {
+    let content = "";
+
+    // Note action.output can be any valid JSON including null.
+    content += `WEBSEARCH OUTPUT:\n`;
+    content += `${JSON.stringify(this.output, null, 2)}\n`;
```
Comment (spolu) [inline front/lib/api/assistant/actions/websearch.ts:84]: (same here)

File: types/src/front/lib/actions/registry.ts [hunk 4]
```diff
@@ -159,7 +159,7 @@ export const DustProdActionRegistry = createActionRegistry({
   "assistant-v2-websearch": {
     app: {
       workspaceId: PRODUCTION_DUST_APPS_WORKSPACE_ID,
-      appId: "78bda07b39",
+      appId: "098b515f8e",
```
Comment (spolu) [inline types/src/front/lib/actions/registry.ts:162]: ?

**Review by spolu — APPROVED** [review]
Left a few questions / comments but LGTM!

File: front/components/assistant/conversation/WebsearchAction.tsx (hunk 1 above)
Comment (philipperolet) [reply front/components/assistant/conversation/WebsearchAction.tsx:1]: the 'dist' is def a typo if that's what you meant :+1: 
the import is necessary to the signature of WebsearchAction (implemented in the following PR)

File: types/src/front/lib/actions/registry.ts (hunk 4 above)
Comment (philipperolet) [reply types/src/front/lib/actions/registry.ts:162]: correcting a mistake from a previous PR (at a time at which the run implementation was not done so app was not called)

File: front/components/assistant/conversation/WebsearchAction.tsx (hunk 1 above)
Comment (spolu) [reply front/components/assistant/conversation/WebsearchAction.tsx:1]: Ah yeah didn't realize it was an edit. But yeah the edit looks fishy also :p

**Pull request merged**